src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -427,11 +427,11 @@ abstract class RefChecks extends Transform {
                                           "(this rule is designed to prevent accidental overrides)")
             } else if (other.isStable && !member.isStable) { // (1.4)
               overrideErrorWithMemberInfo("stable, immutable value required to override:")
-            } else if (member.isValue && member.isLazy &&
+            } else if (member.isValue && (member.isLazy || member.isModule) &&
                        other.isValue && other.hasFlag(STABLE) && !(other.isDeferred || other.isLazy)) {
               overrideErrorWithMemberInfo("concrete non-lazy value cannot be overridden:")
```

Comment:
should we imporove this message?
```
"value must be non-lazy to override a concrete non-lazy value:"
```

Reply:
indeed, cut/paste the next message, add negating prefix.